chainer/variable.py

```diff
@@ -655,8 +655,12 @@ def add_cand(cand):
             cuda.get_device(*(in_data + out_grad)).use()
             for hook in six.itervalues(hooks):
                 hook.backward_preprocess(func, in_data, out_grad)
+            func.output_data = tuple(
+                [None if y is None else y.data for y in outputs])
             gxs = func.backward(in_data, out_grad)
             assert len(gxs) == len(in_data)
+            if not hasattr(func, '_retain_after_backward'):
```

Comment:
will change to use getattr() instead of hasattr(). BTW, is it better not to use hasattr()?